src/store/test/thunk.ts

```diff
@@ -8,3 +8,17 @@ export const fetchTest = createAsyncThunk(
     return test;
   }
 );
+
+export const validateTest = createAsyncThunk(
+  'test/validate',
+  async ({
+    id,
+    answersData,
+  }: {
+    id: number;
+    answersData: TestOnValidationData[];
+  }) => {
+    const test = await lessonsApi.postTest({ id, answersData });
+    return test;
+  }
+);
```

Comment:
Тип уже определен в `api/lessons`, деструктурировать нет необходимости

```suggestion
export const validateTest = createAsyncThunk('test/validate', async (data: AnswersValidationData) => {
    const test = await lessonsApi.postTest(data);
    return test;
  });
```